wynnset/tapestry-wp
# 628 - Enable hiding nodes from Admin and implement Author filter in backend

Closes #628 

Changes: 
 - Add new `superuserOverridesPermission` flag to settings. This is true by default.
 - Add `userId` parameter to `getTapestry` endpoint. If specified, this second user's authored nodes are returned with the current user's, overriding read permissions.
 - Refactored `tapestry-user-roles.php` to no longer be static, and to take a userId argument. Changed all calls to it's functions in the backend to work as objects calls.
- Added `getAllContributors` endpoint to `tapestry.php`. This returns a list of unique authors of nodes within the tapestry.

Steps to test:
1. Login as your tapestry author/editor.
2. Create a node with public add/edit permissions and another with non-public read access.
3. Log out, and log in as another user which is not a super user. 
Superusers are either: Author of the Tapestry, Administator role on WP or Editor role on WP
4. As this non-superuser, you should only be able to see the node with all public permissions.
5. Create a new node as the non-superuser. Turn it's public read permissions off, so only that user can view it.
6. Log back into the admin account and verify the superuser can see all 3 nodes in the tapestry
7. Go to settings and flip the `superuserOverridesPermission` flag to false
8. After reload, verify the superuser can no longer see the 2nd account's node. Check the backend and ensure the visible nodes are the only ones within the dataset.
9. Activate the filter, and select the 2nd user's node as filtered
10. Ensure the 2nd users nodes appear along with the 1st.
11. Deactivate the filter and ensure the nodes go back to the original state. Check the backend and ensure the visible nodes are the only ones within the dataset.
